crates/engine_bibtex/bibtex/bibtex.c

```diff
@@ -529,8 +529,16 @@ input_ln(peekable_input_t *peekable)
         buffer[last] = peekable_getc(peekable);
         last++;
     }
+    
+    // For side effects - consume the eoln we saw
+    int eoln = peekable_getc(peekable);
 
-    peekable_getc(peekable);
+    // Handle \r\n newlines on Windows by trying to consume a \n after a \r, unget if it's not that exact pair
+    int next = peekable_getc(peekable);
+    printf("Next Char: %d\n", next);
+    if (!(eoln == '\r' && next == '\n')) {
```

Comment:
Two items here:

1. Looks like there is a debug `printf` that should be removed.
2. I'm not sure if it would be any more efficient or anything, but I think I'd rather have the code here only attempt the second getc if the first character is `\r`. I think that makes the internal logic a bit clearer.